rgw/admin/info.go

```diff
@@ -0,0 +1,37 @@
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+/* parse a nested json expression like
+ * {"info":{"storage_backends":[{"name":"rados","cluster_id":"75d1938b-2949-4933-8386-fb2d1449ff03"}]}} */
+
+// Info struct
+type Backend struct {
+	Name    string `json:"name"`
+	ClusterID    string `json:"cluster_id"`
+}
+
+type Info struct {
+	Info map[string] []Backend
```

Comment:
I don't think this is a valid type. This works for me though:

```
	type StorageBackend struct {
		Name      string `json:"name"`
		ClusterID string `json:"cluster_id"`
	}

	type Info struct {
		InfoSpec struct {
			StorageBackends []StorageBackend `json:"storage_backends"`
		} `json:"info"`
	}

	info := []byte(`{"info":{"storage_backends":[{"name":"rados","cluster_id":"75d1938b-2949-4933-8386-fb2d1449ff03"}]}}`)

	u := Info{}
	err := json.Unmarshal(info, &u)
	assert.NoError(t, err)
	assert.Equal(t, "rados", u.InfoSpec.StorageBackends[0].Name)
```

Reply:
go 1.14 compiled it, but I make no claims for it--just needed something to test with.
Do you want to take this over @leseb ?